src/Orleans/Lifecycle/ILifecycleObserver.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System.Threading.Tasks;
+
+namespace Orleans
+{
+    /// <summary>
+    /// Lifecycle observer used to handle start and stop notification.
+    /// </summary>
+    public interface ILifecycleObserver
+    {
+        /// <summary>
+        /// Handle start notifications
+        /// </summary>
+        /// <returns></returns>
+        Task OnStart();
+
+        /// <summary>
+        /// Handle stop notifications
+        /// </summary>
+        /// <returns></returns>
+        Task OnStop();
+    }
```

Comment:
I don't foresee complex subscription management being a common practice in this space, but let's see what usage patterns evolve.  My preference would be to keep these interfaces minimalistic until more complexity is needed.